### PR TITLE
Add support for Maximum to TensorFlowLite.

### LINKED ISSUE #14661: TensorFlowMaximum operator in TF Lite
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Docker 
- **TensorFlow version (use command below)**: 1.4.0
- **Python version**: 3.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NA
- **Exact command to reproduce**:

### Describe the problem
I am trying to convert a graph from .pb to .lite format using toco, but I get this error:
`2017-11-17 10:20:47.738777: F tensorflow/contrib/lite/toco/tflite/export.cc:192] Unsupported operator: TensorFlowMaximum`

I had a look inside the code, and found `TensorFlowMaximum` operator defined inside `lite/kernels/internal/optimized/optimized_ops.h`, but apparently it's not linked or defined anywhere else.

What's the fastest way to get this defined and linked?

Thanks a lot.

### PR DESCRIPTION
**Pull request by StanislawAntol**
I don't have that many tests (e.g., not sure if broadcasting works), but running `bazel test tensorflow/contrib/lite/kernels:maximum_test`, `bazel test tensorflow/contrib/lite/toco/tflite:operator_test`, and `bazel test //tensorflow/contrib/lite/testing:generated_examples_zip_test` all passed.

To help close #14661.

### CODE REVIEW

File: tensorflow/contrib/lite/kernels/maximum.cc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      GetTensorDims(op_context.output))
+
+  if (kernel_type == kReference) {
+    switch (op_context.output->type) {
+      case kTfLiteFloat32:
+        TF_LITE_MAXIMUM(reference_ops, float);
+        break;
+      default:
+        context->ReportError(context,
+                             "Type %d is currently not supported by TFMaximum.",
```
Comment (andrehentz) [inline tensorflow/contrib/lite/kernels/maximum.cc]: You might need to fix the formatting here, so everything fits in 80 columns.

File: tensorflow/contrib/lite/testing/generate_examples.py [hunk 2]
```diff
@@ -780,6 +780,39 @@ def build_inputs(parameters, sess, inputs, outputs):
...
+        name="input_2",
+        shape=parameters["input_shape_2"])
+
+    out = tf.maximum(input_tensor_1, input_tensor_2)
+    return [input_tensor_1, input_tensor_2], [out]
+
+  def build_inputs(parameters, sess, inputs, outputs):
+    values = [
+        create_tensor_data(parameters["input_dtype"], parameters["input_shape_1"]),
+        create_tensor_data(parameters["input_dtype"], parameters["input_shape_2"])
```
Comment (andrehentz) [inline tensorflow/contrib/lite/testing/generate_examples.py]: make sure everything fits in 80 cols.

**Review by andrehentz — APPROVED** [review]

File: tensorflow/contrib/lite/kernels/maximum.cc (hunk 1 above)
Comment (StanislawAntol) [reply tensorflow/contrib/lite/kernels/maximum.cc]: Fixed these and the merge conflicts in the `schema` files.

**Comment by ekelsen** [comment]
Can you please resolve the conflicts?

**Comment by StanislawAntol** [comment]
@ekelsen , I have tried to update it and am currently running the tests, but it seems to be failing to due some other recent changes with `make_lstm_tests` (i.e., I just tried the same test on the `tensorflow/tensorflow` master banch):

```
Traceback (most recent call last):
  File "/home/s.antol/.cache/bazel/_bazel_s.antol/d5f2817d0f58daa7ca384d9ca065c3f9/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/contrib/lite/testing/generate_examples.runfiles/org_tensorflow/tensorflow/contrib/lite/testing/generate_examples.py", line 1950, in <module>
    tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
  File "/home/s.antol/.cache/bazel/_bazel_s.antol/d5f2817d0f58daa7ca384d9ca065c3f9/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/contrib/lite/testing/generate_examples.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 126, in run
    _sys.exit(main(argv))
  File "/home/s.antol/.cache/bazel/_bazel_s.antol/d5f2817d0f58daa7ca384d9ca065c3f9/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/contrib/lite/testing/generate_examples.runfiles/org_tensorflow/tensorflow/contrib/lite/testing/generate_examples.py", line 1936, in main
    dispatch[out](_path(out))
  File "/home/s.antol/.cache/bazel/_bazel_s.antol/d5f2817d0f58daa7ca384d9ca065c3f9/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/contrib/lite/testing/generate_examples.runfiles/org_tensorflow/tensorflow/contrib/lite/testing/generate_examples.py", line 1839, in make_lstm_tests
    use_frozen_graph=True)
  File "/home/s.antol/.cache/bazel/_bazel_s.antol/d5f2817d0f58daa7ca384d9ca065c3f9/execroot/org_tensorflow/bazel-out/host/bin/tensorflow/contrib/lite/testing/generate_examples.runfiles/org_tensorflow/tensorflow/contrib/lite/testing/generate_examples.py", line 494, in make_zip_of_tests
    "Found %d errors while generating toco models" % toco_errors)
RuntimeError: Found 1 errors while generating toco models
Target //tensorflow/contrib/lite/testing:generated_examples_zip_test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 366.084s, Critical Path: 92.76s
FAILED: Build did NOT complete successfully

Executed 0 out of 1 test: 1 fails to build.
```

Any suggestions?

**Comment by andrehentz** [comment]
@StanislawAntol  Anything I can help with?

It looks like StringMapList is not supported outside of google. Sorry about that. It may be easiest to just disable the make_lstm_tests.

**Comment by StanislawAntol** [comment]
I just pushed some commits that fixed any merge issues and pass the tests (commenting out the LSTM one).

**Comment by StanislawAntol** [comment]
@yifeif , @andrehentz 
Is there anything I can do to speed up getting this merged into master so it does not need another set of merge conflicts?

**Comment by yifeif** [comment]
infra issue, rerunning

**Comment by frankcarey** [comment]
I'm having trouble running a model converted with the code above to a tflite after modifying the tflite demo app for android (also a tiny-yolo model).  I'm guessing that I need to use the same tf java and c++ code when I build the app as when I converted the model?

**Comment by StanislawAntol** [comment]
@frankcarey , yes, you will need to update the build system such that it includes and builds the maximum.cc file (we were building for Android via CMake).

**Comment by andrehentz** [comment]
@StanislawAntol Can I do anything to help here?

**Comment by StanislawAntol** [comment]
@andrehentz , I've just been a bit busy at my new job and haven't gotten around to fixing the conflicts and dealing with the CLA issue.

**Comment by StanislawAntol** [comment]
@andrehentz , I updated everything and the 3 tests are passing for me.

**Review by andrehentz — APPROVED** [review]

**Comment by StanislawAntol** [comment]
@andrehentz , @yifeif Can one of you launch the build checks?

**Pull request merged**